distuv/chisquared.go

```diff
@@ -0,0 +1,74 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+)
+
+// ChiSquared implements the ChiSquared distribution, a one parameter distribution
+// with support on the positive numbers.
+//
+// The density function is given by
+//  1/(2^{k/2} * Γ(k/2)) * x^{k/2 - 1} * e^{-x/2}
+// It is a special case of the Gamma distribution, Γ(k/2, 1/2).
+//
+// For more information, see https://en.wikipedia.org/wiki/Chi-squared_distribution.
+type ChiSquared struct {
+	K float64 // Shape parameter, must be greater than 0.
```

Comment:
Add a note that this corresponds to degrees of freedom? I anticipate that this function will see a lot of undergrads looking for ways to do tests, and not knowing what K is or how to specify dof.